cli/src/semgrep/commands/install.py

```diff
@@ -27,8 +27,15 @@
 
 logger = getLogger(__name__)
 
+SEMGREP_PRO_PATH = os.getenv("SEMGREP_PRO_PATH")
+
 
 def determine_semgrep_pro_path() -> Path:
+    if SEMGREP_PRO_PATH:
```

Comment:
Unfortunately we have some duplicated code. Could you also update `find_semgrep_core_path` in `semgrep/cli/bin`? Or at least leave a note

Reply:
Do we need to modify that function? That function looks for binaries in PATH and we've made sure to put the location we're storing the pro binary (/home/semgrep/bin) into PATH in the Dockerfile.

Reply:
To rephrase: If a user sets the `SEMGREP_PRO_PATH` env var to a location that's not on `PATH`, runs the install to put the pro binary in that directory, and then runs the cli, it seems like `find_semgrep_core_path` will correctly raise a `CoreNotFound` error and explain it couldn't find the binary in `PATH`. This behavior seems consistent with the user experience in other cases like installing semgrep OSS with Homebrew if the brew directory isn't in `PATH`, so it's not clear to me what modifications we need to make to `find_semgrep_core_path`.

Reply:
@emjin at first i also thought this file/code might have to be modified as well, but in looking at it, i agree with @cgdolan , all `find_semgrep_core_path` does is locate the path of the binary, not set where to install it.

Reply:
Ah, I see, I missed that `PATH` was set! In that case, this is fine without. I would still prefer a comment, because I think it would be confusing for a local `semgrep scan` user who had set `SEMGREP_PRO_PATH` but was getting an error that semgrep-pro isn't found, but it's not urgent to implement

Reply:
@emjin sorry I don't exactly follow what you mean 😅 . would it be clearer if we changed `SEMGREP_PRO_PATH` to `SEMGREP_PRO_INSTALL_PATH`?

Reply:
What I'm concerned about is:
1. A user doesn't have semgrep pro
2. They set SEMGREP_PRO_PATH and run `semgrep ci --pro`
3. `semgrep ci --pro` installs semgrep pro for the user
4. They later run `semgrep scan --pro`, which cannot find semgrep pro. The user is confused

It's not a super common scenario, so I'm not that worried, but it seems like we could avoid this by also looking for the pro binary in SEMGREP_PRO_PATH

Let me know if I've misunderstood something!

Reply:
I totally get what you mean now, sorry it took me a minute to realize the scenario. I added logic for `semgrep scan --pro` to look for the pro binary in SEMGREP_PRO_PATH if it's set. I also added some more error handling in general and code comments.